src/models/__tests__/service.test.ts

```diff
@@ -166,3 +181,113 @@ describe("listLastVersionServices", () => {
+    );
+    const model = new ServiceModel(containerMock);
+    const result = await model.create({...aRawServiceWithMetadata, kind: "INewService"})();
+    expect(mockCreate).toBeCalled();
+    expect(E.isRight(result)).toBeTruthy();
+    if (E.isRight(result)) {
+      expect(result.right).toEqual({...aRetrievedService, serviceMetadata, kind: "IRetrievedService", id: expect.any(String)})
+    }
+  });
+})
+
+describe("Special Service metadata types", () => {
+
+  it("should decode services metadata without category as Standard Services metadata", async () => {
+    const oldServiceMetadata: CommonServiceMetadata = {
+      scope: ServiceScopeEnum.LOCAL
+    };
+    const asyncIterable = getAsyncIterable([[{...aRetrievedService, serviceMetadata: oldServiceMetadata}]]);
+    mockGetAsyncIterator.mockReturnValueOnce(asyncIterable);
+
+    const model = new ServiceModel(containerMock);
+
+    const result = await model.listLastVersionServices()();
+
+    expect(E.isRight(result)).toBeTruthy();
+    if (E.isRight(result)) {
+      const values = O.toUndefined(result.right);
+      expect(values).toHaveLength(1);
+      if (values !== undefined) {
+        expect(values[0]).toHaveProperty("serviceMetadata.category", StandardServiceCategoryEnum.STANDARD);
+      }
+    }
+  });
+
+  it("should decode services metadata with category", async () => {
+    const standardServiceMetadata: StandardServiceMetadata = {
+      scope: ServiceScopeEnum.LOCAL,
+      category: StandardServiceCategoryEnum.STANDARD,
+      customSpecialFlow: undefined
+    };
+
+    const specialServiceMetadata: SpecialServiceMetadata = {
+      scope: ServiceScopeEnum.LOCAL,
+      category: SpecialServiceCategoryEnum.SPECIAL,
+      customSpecialFlow: "custom-flow-name" as NonEmptyString
+    };
+
+    const anotherService = {...aRetrievedService, serviceId: "anotherServiceId"};
+
+    const asyncIterable = getAsyncIterable([[{...aRetrievedService, serviceMetadata: standardServiceMetadata}, {...anotherService, serviceMetadata: specialServiceMetadata}]]);
+    mockGetAsyncIterator.mockReturnValueOnce(asyncIterable);
+
+    const model = new ServiceModel(containerMock);
+
+    const result = await model.listLastVersionServices()();
+
+    expect(E.isRight(result)).toBeTruthy();
+    if (E.isRight(result)) {
+      const values = O.toUndefined(result.right);
+      expect(values).toHaveLength(2);
+      if (values !== undefined) {
+        expect(values[0]).toHaveProperty("serviceMetadata", standardServiceMetadata);
+        expect(values[1]).toHaveProperty("serviceMetadata", specialServiceMetadata);
+      }
+    }
+  });
+
+  it("should require category on NewService type", () => {
+
+    const aCommonServiceMetadata: CommonServiceMetadata = {
+      scope: ServiceScopeEnum.LOCAL
+    };
+
+    const aNewServiceWithoutCategory: NewService = {
+      ...aRawService,
+      kind: "INewService",
+      // @ts-expect-error
+      serviceMetadata: aCommonServiceMetadata
+    }
+
```

Comment:
I was expecting this code be correct for new services to be created.

```typescript
const decodedValue = NewService.decode(aNewServiceWithoutCategory);
expect(E.isLeft(decodedValue)).toBeTruthy();
```

Reply:
The decode should succeeded because the missing category in serviceMetadata has a `withDefault` to STANDARD.
The right code should be
```typescript
const aNewServiceWithoutCategory: NewService = {
      ...aRawService,
      kind: "INewService",
      // @ts-expect-error
      serviceMetadata: aCommonServiceMetadata
    }
const decodedValue = NewService.decode(aNewServiceWithoutCategory);
expect(E.isRight(decodedValue)).toBeTruthy();
```
Anyway the behavior of this test is verify the type-check.

Reply:
The previous checks was added in b040339

Reply:
I understand that retrieving an old service from the DB needs a default STANDARD decoder.
But I thought that when adding a new Service, that parameter would become mandatory.